internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -61,19 +64,24 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "start reconcile", req.NamespacedName.String())
 	defer level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "end reconcile", req.NamespacedName.String())
 
-	svcName := strings.TrimSuffix(req.Name, fmt.Sprintf("-%s", r.NodeName))
-	ipAdvS := r.StatusFetcher(types.NamespacedName{Name: svcName, Namespace: req.Namespace})
+	serviceName, serviceNamespace := req.Name, req.Namespace
+	statusName, statusNamespace := fmt.Sprintf("%s-%s-%s", req.Name, req.Namespace, r.NodeName), r.Namespace
```

Comment:
nit: use serviceName as the argument in the formatting string will make it clearer how the statusName is built with.
Also, @oribon had concerns about generating such name in this way with the risk of going over the maximum size. The alternative would be to use a generated name, but doing so would make the status - service / node binding implemented only with something immutable like labels.

If we do so, we'd risk to have somebody changing the label and leaving a rogue status around. A possible alternative would be:
- generated name
- node and service name as labels (for better discoverability)
- node and service name explicit in the status part (to provide immutability of the information)

If we do that, we'll probably have to add an index as well so we can retrieve the entry easily (we won't be able to do that by name anymore).

Thoughts?

Based on the outcome of this discussion, we'll have to choose what to do with the predicate.

Reply:
What do you mean by adding an index? Can you give me an quick example for better understanding?

Reply:
Something along the line of what we do here https://github.com/metallb/metallb/blob/main/internal/k8s/k8s.go#L246 where we set a key to retrieve the ep slice based on the service name

Reply:
+1 for that idea!

Reply:
Let me try to tidy this up.
With completely generated names for crs, along with index which maintains the node->svc relationship, we would    use something like this in predicate to determine if a cr belongs to a specific node(speaker), am I right ?
https://github.com/metallb/metallb/blob/83b5dd4739e32d22348524fbf7ffedc4f1a7ad1f/internal/k8s/controllers/endpoints.go#L16-L18

Reply:
Thought some details and found there may be a tiny problem with the nodeName.
When the first time a status cr is created, there is no status sub-resource. The status sub-resource is patched after a re-enqueue.
So It seems not possible to implement our predicate based on something like status.nodeName(metadata.name/metadata.labels seems more applicable).

Reply:
Will submit this change this weekend 😄